packager/media/codecs/aac_audio_specific_config.cc

```diff
@@ -162,6 +162,33 @@ bool AACAudioSpecificConfig::ConvertToADTS(std::vector<uint8_t>* buffer) const {
   return true;
 }
 
+std::vector<uint8_t> AACAudioSpecificConfig::GetAACwithADTSHeader(const uint8_t* data, size_t length) const {
+  DCHECK(audio_object_type_ >= 1 && audio_object_type_ <= 4 &&
+         frequency_index_ != 0xf && channel_config_ <= 7);
+
+  std::vector<uint8_t> aac_with_adts_header(data, data + length);
```

Comment:
There are still two copies (line 169 and line 179), which is identical to the original code. 

You can remove the extra copying by appending the data later, e.g.

std::vector<uint8_t> aac_with_adts_header;
aac_with_adts_header.reserve(length +  kADTSHeaderSize);
...
aac_with_adts_header.insert(aac_with_adts_header.end(), kADTSHeaderSize, 0);
...
aac_with_adts_header.insert(aac_with_adts_header.end(), data, data+length);